src/main/java/org/codehaus/plexus/archiver/util/ArchiveEntryUtils.java

```diff
@@ -17,186 +17,64 @@
-                return;
-            }
-            catch ( IOException e )
-            {
-                throw new ArchiverException( "Failed setting file attributes with java7+", e );
-            }
-        }
-
-        final String m = Integer.toOctalString( mode & 0xfff );
-
-        if ( useJvmChmod && !jvmFilePermAvailable )
-        {
-            useJvmChmod = false;
-        }
-
-        if ( useJvmChmod )
-        {
-            applyPermissionsWithJvm( file, m, logger );
-            return;
-        }
-
-        try
-        {
-            final Commandline commandline = new Commandline();
-
-            commandline.setWorkingDirectory( file.getParentFile().getAbsolutePath() );
-
-            if ( logger.isDebugEnabled() )
-            {
-                logger.debug( file + ": mode " + Integer.toOctalString( mode ) + ", chmod " + m );
-            }
-
-            commandline.setExecutable( "chmod" );
-
-            commandline.createArg().setValue( m );
-
-            final String path = file.getAbsolutePath();
-
-            commandline.createArg().setValue( path );
-
-            // commenting this debug statement, since it can produce VERY verbose output...
-            // this method is called often during archive creation.
-            // logger.debug( "Executing:\n\n" + commandline.toString() + "\n\n" );
-            final CommandLineUtils.StringStreamConsumer stderr = new CommandLineUtils.StringStreamConsumer();
-
-            final CommandLineUtils.StringStreamConsumer stdout = new CommandLineUtils.StringStreamConsumer();
-
-            final int exitCode = CommandLineUtils.executeCommandLine( commandline, stderr, stdout );
-
-            if ( exitCode != 0 )
-            {
-                logger.warn( "-------------------------------" );
-                logger.warn( "Standard error:" );
-                logger.warn( "-------------------------------" );
-                logger.warn( stderr.getOutput() );
-                logger.warn( "-------------------------------" );
-                logger.warn( "Standard output:" );
-                logger.warn( "-------------------------------" );
-                logger.warn( stdout.getOutput() );
-                logger.warn( "-------------------------------" );
-
-                throw new ArchiverException( "chmod exit code was: " + exitCode );
-            }
-        }
-        catch ( final CommandLineException e )
-        {
-            throw new ArchiverException( "Error while executing chmod.", e );
-        }
-
+        chmod( file, mode );
     }
 
     /**
-     * <b>jvm chmod will be used only if System property <code>useJvmChmod</code> set to true</b>
+     * This method is now deprecated.
      *
-     * @param file
-     * @param mode
-     * @param logger
+     * The {@code logger} provided is no longer used.
      *
-     * @throws ArchiverException
+     * @deprecated Use {@link #chmod(File, int)}
      */
+    @Deprecated
```

Comment:
This method is obsolete but since it's public I guess it's better to keep it for now (maybe until the next major release?).